presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDriver.java

```diff
@@ -46,7 +54,8 @@
 
     private static final String USER_PROPERTY = "user";
 
-    private final QueryExecutor queryExecutor;
+    private final LoadingCache<Properties, QueryExecutor> queryExecutorCache;
+    private final ReadWriteLock cacheLock;
```

Comment:
You don't need this lock. `LoadingCache` is thread safe. And as i see you don't need the entire `closer` concept. You can close the entire `QueryExecutor` upon expiration. It closes the `HttpClient` internally.